### DISCUSSION TITLE
[v4] Custom breakpoints paddings are not respected

### DISCUSSION THREAD

**Discussion by pavelbelyaev98** [tailwindlabs/tailwindcss#15450] [Help] [answered]
When we have a following class:

```px-4 lg:px-6 3xl:px-20```

The 3xl breakpoint will NOT be respected, instead the lg breakpoint will get the priority, even thought it should use the 3xl breakpoint.

![image](https://github.com/user-attachments/assets/654ef7f3-8310-4c02-ac57-7973a5a86475)
![image](https://github.com/user-attachments/assets/0b2d11c3-0f53-42cb-8c16-54f82d14f6a7)

NOTE: this happens only for padding, it seems fine for background colors.

With the following theme config:

```
@theme {
  --breakpoint-3xl: 1600px;
}
```

As a workaround i have to specify that 3xl should take priority with the "!" sign:

```px-4 lg:px-6 3xl:px-20!```

**What version of Tailwind CSS are you using?**

^4.0.0-beta.8

**What build tool (or framework if it abstracts the build tool) are you using?**

Nextjs 15.1

**What version of Node.js are you using?**

22

**What browser are you using?**

Edge

**What operating system are you using?**

Windows

**Reproduction URL**
https://play.tailwindcss.com/1tQGhhPGfR

**Describe your issue**

Describe the problem you're seeing, any important steps to reproduce and what behavior you expect instead.

**Comment by philipp-spiess** [comment, marked as answer]
Hey! We're sorting by the breakpoint value when generating these classes, the issue you're seeing is because you're adding a `px` value where as the default breakpoints are configured in `rem` instead. This means the values can't be sorted correctly anymore because of the unit mismatch.

I recommend setting the custom breakpoint to `rem` as well so it works as expected: https://play.tailwindcss.com/Zw7yHqbReM

**Reply by pavelbelyaev98** [reply]
Thanks!

**Reply by valgeirb** [reply]
This is crucial information that should be mentioned in the docs/upgrade guide in my opinion.

I'm glad I found this question as I was having the exact same issue.

**Reply by valgeirb** [reply]
You even use `px` in your [examples](https://tailwindcss.com/docs/adding-custom-styles#customizing-your-theme)

**Reply by abdullah-dev-mamun** [reply]
please use rem instead of px , That will work.
